services/web/server/src/simcore_service_webserver/projects/projects_nodes_handlers.py

```diff
@@ -99,6 +101,71 @@ async def get_node(request: web.Request) -> web.Response:
         raise web.HTTPNotFound(reason=f"Project {project_uuid} not found") from exc
 
 
+@routes.get(f"/{VTAG}/projects/{{project_id}}/nodes/{{node_id}}/resources")
+@login_required
+@permission_required("project.node.read")
+async def get_node_resources(request: web.Request) -> web.Response:
+    user_id: UserID = request[RQT_USERID_KEY]
+
+    try:
+        project_uuid = ProjectID(request.match_info["project_id"])
+        node_uuid = NodeID(request.match_info["node_id"])
+    except KeyError as err:
+        raise web.HTTPBadRequest(reason=f"Invalid request parameter {err}") from err
+    try:
+        # ensure the project exists
+        project = await projects_api.get_project_for_user(
+            request.app,
+            project_uuid=f"{project_uuid}",
+            user_id=user_id,
+            include_templates=True,
+        )
+
+        resources = await projects_api.get_project_node_resources(
+            request.app, project=project, node_id=node_uuid
+        )
+        return web.json_response({"data": resources}, dumps=json_dumps)
+
+    except ProjectNotFoundError as exc:
+        raise web.HTTPNotFound(reason=f"Project {project_uuid} not found") from exc
+    except NodeNotFoundError as exc:
+        raise web.HTTPNotFound(reason=f"Node {node_uuid} not found in project") from exc
+
+
+@routes.put(f"/{VTAG}/projects/{{project_uuid}}/nodes/{{node_uuid}}/resources")
+@login_required
+@permission_required("project.node.update")
+async def replace_node_resources(request: web.Request) -> web.Response:
+    user_id: UserID = request[RQT_USERID_KEY]
+
+    try:
+        project_uuid = ProjectID(request.match_info["project_id"])
+        node_uuid = NodeID(request.match_info["node_id"])
+        _body = await request.json()
+    except KeyError as err:
+        raise web.HTTPBadRequest(reason=f"Invalid request parameter {err}") from err
+    try:
+
+        # ensure the project exists
+        _project = await projects_api.get_project_for_user(
+            request.app,
+            project_uuid=f"{project_uuid}",
+            user_id=user_id,
+            include_templates=True,
+        )
+        raise web.HTTPNotImplemented(reason="Not yet implemented!")
+        # new_node_resources = await projects_api.set_project_node_resources(
```

Comment:
No need to comment his. We have this already in place: a middleware that maps NotImplementedErrors into web.HTTPNotImplemented. 

SEE https://github.com/pcrespov/osparc-simcore/blob/master/packages/service-library/src/servicelib/aiohttp/rest_middlewares.py#L120

Reply:
nope. it fails.

Reply:
what do you mean? the middleware does not work?

Reply:
actually the tests are failing. so for now I'll keep it so since this will anyway change with the next iteration

Reply:
the test that checks the assertion does it wrong on the test side. cause I test for the error at the moment... we can check again tomorrow, but that's a minor issue.